src/browser/modules/Sidebar/ProjectFiles.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2002-2020 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import React, { useEffect, useState } from 'react'
+import { connect } from 'react-redux'
+import { withBus } from 'react-suber'
+import { Bus } from 'suber'
+import { useMutation } from '@apollo/client'
+
+import Render from 'browser-components/Render'
+import { Drawer, DrawerHeader } from 'browser-components/drawer'
+import ProjectFilesScripts, { ProjectFilesError } from './ProjectsFilesScripts'
+import {
+  StyledSetting,
+  StyledSettingLabel,
+  StyledSettingTextInput
+} from './styled'
+import {
+  setProjectFileDefaultFileName,
+  updateCacheAddProjectFile
+} from './project-files.utils'
+import { SET_CONTENT } from 'shared/modules/editor/editorDuck'
+import { CYPHER_FILE_EXTENSION } from 'shared/services/export-favorites'
+import {
+  SELECT_PROJECT_FILE,
+  SAVE_PROJECT_FILE,
+  ADD_PROJECT_FILE,
+  PROJECT_FILES_MOUNTED,
+  PROJECT_FILES_UNMOUNTED
+} from './project-files.constants'
+
+interface IProjectFiles {
```

Comment:
Perhaps ProjectFilesProps is a better name?

Reply:
will take a look

Reply:
I think it's a good idea to prefix interfaces as they behave differently than pure types. It's a well established convention in the broader programming community IMO

Reply:
deferring here to the other thread on naming